crates/primitives-traits/src/extended.rs

```diff
@@ -9,12 +9,14 @@ use alloy_primitives::{bytes::Buf, ChainId, TxHash};
 use alloy_rlp::{BufMut, Decodable, Encodable, Result as RlpResult};
 use op_alloy_consensus::OpTxEnvelope;
 use reth_codecs::Compact;
-use reth_ethereum::primitives::{
-    serde_bincode_compat::SerdeBincodeCompat, transaction::signed::RecoveryError, InMemorySize,
-    SignedTransaction,
-};
 use revm_primitives::{Address, Bytes, TxKind, B256, U256};
 
+use crate::{
+    serde_bincode_compat::SerdeBincodeCompat,
```

Comment:
this impl we need to feature gate, with https://github.com/paradigmxyz/reth/blob/22452e8ace68ed19ebe7110d2cd34a94617c8ec5/crates/primitives-traits/Cargo.toml#L119-L119